src/python/director/lcmlogplayer.py

```diff
@@ -142,29 +149,71 @@ def __init__(self, logPlayer):
         stopButton = QtGui.QPushButton('Stop')
         slider = QtGui.QSlider(QtCore.Qt.Horizontal)
         slider.maximum = int(logPlayer.getEndTime()*100)
+        text = QtGui.QLineEdit()
+        text.text = '0.0'
         playButton.connect('clicked()', self.onPlay)
         stopButton.connect('clicked()', self.onStop)
         slider.connect('valueChanged(int)', self.onSlider)
+        text.connect('returnPressed()', self.onText)
 
         l = QtGui.QHBoxLayout(w)
         l.addWidget(slider)
+        l.addWidget(text)
         l.addWidget(playButton)
         l.addWidget(stopButton)
 
         self.slider = slider
+        self.text = text
         self.widget = w
         self.widget.show()
 
+    def _getSliderTime(self, value=None):
+        if value is None:
+            value = self.slider.value
+        t = self.logPlayer.getEndTime()*value/self.slider.maximum
+        return t
+
+    def _getSliderValue(self, t):
+        value = t / self.logPlayer.getEndTime() * self.slider.maximum
+        return int(round(value))
+
+    def _updateTime(self, t):
+        with BlockSignals(self.slider, self.text):
+            self.slider.value = self._getSliderValue(t)
+            self.text.text = str(t)
+
     def onPlay(self):
-        self.logPlayer.playback(0.0, self.logPlayer.getEndTime())
+        self.logPlayer.playback(self._getSliderTime(), self.logPlayer.getEndTime(), self._updateTime)
 
     def onStop(self):
         self.logPlayer.timer.stop()
 
-    def onSlider(self, value):
-        t = self.logPlayer.getEndTime()*value/self.slider.maximum
+    def skipTo(self, t):
+        self._updateTime(t)
         self.logPlayer.skipToTime(t, playLength=0.0)
 
+    def onSlider(self, value):
+        t = self._getSliderTime(value)
+        self.skipTo(t)
+
+    def onText(self, *args):
+        try:
+            t = float(self.text.text)
+            self.skipTo(t)
+        except ValueError:
+            pass
+
+# @ref https://stackoverflow.com/a/35000974/7829525
+class BlockSignals(object):
```

Comment:
awesome, can you put this class into `qtutils.py`

https://github.com/RobotLocomotion/director/blob/master/src/python/director/qtutils.py